specification/compute/resource-manager/Microsoft.Compute/stable/2021-10-01/examples/gallery/CreateOrUpdateASimpleGalleryImageVersionWithTargetExtendedLocations.json

```diff
@@ -0,0 +1,323 @@
+{
+  "parameters": {
+    "subscriptionId": "{subscription-id}",
```

Comment:
Can you explain how this works with the targetExtendedLocation ?

Looks like you create the resource and the targetExtendedLocations automatically get set as per this example. Also rather than using the extendedlocation name can you consider using the extendedlocation resourceid instead ?

Reply:
@kangsun-ctrl

Reply:
@raosuhas  Not sure I understand your question correctly here. 'targetExtendedLocations' is used just like 'targetRegions', so it is not automatically set, it is from user input, user needs to input extendedLocation which is edgeZone alongside with 'targetRegions'.


It has a legit payload:
            "targetExtendedLocations": [
              {
                "name": "West US",
                "extendedLocation": {
                  "name": "microsoftlosangeles1",
                  "type": "EdgeZone"
                },
                "encryption": {
                  "osDiskImage": {
                    "diskEncryptionSetId": "/subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/diskEncryptionSet/myWestUSDiskEncryptionSet"
                  },
                  "dataDiskImages": [
                    {
                      "lun": 0,
                      "diskEncryptionSetId": "/subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/diskEncryptionSet/myWestUSDiskEncryptionSet"
                    }
                  ]
                },
                "storageAccountType": "StandardSSD_LRS(default)",
                "extendedLocationReplicaCount": 1
              },
              {
                "name": "East US",
                "extendedLocation": {
                  "name": "microsoftnewyork1",
                  "type": "EdgeZone"
                },
                "encryption": {
                  "osDiskImage": {
                    "diskEncryptionSetId": "/subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/diskEncryptionSet/myEastUSDiskEncryptionSet"
                  },
                  "dataDiskImages": [
                    {
                      "lun": 0,
                      "diskEncryptionSetId": "/subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/diskEncryptionSet/myEastUSDiskEncryptionSet"
                    }
                  ]
                },
                "storageAccountType": "StandardSSD_LRS(default)",
                "extendedLocationReplicaCount": 1
              }
            ],